### ISSUE TITLE
Online with Multiple ToxIDs

### ISSUE THREAD

**Issue by srkunze** [irungentoo/toxcore#1360]
I started the discussion about handling multiple profiles within a single qTox application.
cf. https://github.com/tux3/qTox/issues/1018 and 

It turned out, unfortunately, that the current tox core does not provide the proper API in order to retain compatibility when qTox would implement such a thing like being online with multiple profiles at the same time.
cf. https://github.com/tux3/qTox/issues/1018#issuecomment-108449695 and https://github.com/tux3/qTox/issues/1018#issuecomment-108584970

So, this issue is about whether toxcore is going to provide some standardized interface for this in the future or whether clients need to find their own solutions (which might stall development in this field due to missing consent).

Every outcome is fine, just asking about what's planned for the future. :)

**Comment by 4311021x** [comment]
I agree. Support for multiple profiles would be great!

**Comment by Jeeppler** [comment]
It would be nice to have one master key and than subkeys for the physical nodes. 
The master key is the first node the user creates, all other physical nodes have to be accepted and have to be verified by the node with the master key. 

It is just an simple idea, maybe someone has a better idea and more knowledge of how to implement such a thing. But it is not good that a profile is bound on a specific physical device. It should be possible to use tox even for inter device communication (e. g. mobile phone to laptop and vice versa)

**Comment by ProMcTagonist** [comment]
The nodes should be equal to each other for less headache.

**Comment by Jeeppler** [comment]
@ProMcTagonist this is true you are right, my idea was not that good, but the problem still exists. 
Another idea: Syncthing, the Mozilla Firefox Sync service and Blue-tooth do a device pairing or peer pairing. This is also done by the pidgin OTR plugin. It works like that (extremely simplified): 
1. Device A generates a Secret
2. The secret will be send to device B
3. The user of B types the secret into another field
4. Device A gets the secret back
now the communication is established and both A and B synchronize information.

Don't get me wrong this is just the user perspective.

**Comment by GrayHatter** [comment]
this isn't a full proposal so it's just a dup of #843

**Comment by srkunze** [comment]
@GrayHatter Not sure why you came to this conclusion but it's not about multiple accounts spread across devices and about syncing contacts at all.

It's about running a single instance of tox and being online with multiple ids through that instance.

Did you read the linked comments where qtox dev express their concern because of the limitations of the core?

**Comment by GrayHatter** [comment]
Any reason you're unable to start and use multiple Tox instances?

**Comment by srkunze** [comment]
Usability, familiarity with other clients and uncertainty of what qtox will do to my profile files when run in parallel.
